src/app/src/test/kotlin/org/imperial/mrc/hint/unit/controllers/ADRControllerTests.kt

```diff
@@ -358,6 +360,54 @@ class ADRControllerTests : HintrControllerTests()
         assertThat(result.body!!).isEqualTo("whatever")
     }
 
+    @Test
+    fun `does not update ADR if uploaded file has no changes`()
+    {
+        val hash = mapOf("hash" to "D41D8CD98F00B204E9800998ECF8427E")
+        val data = mapOf("data" to hash)
+        val mockAPIClient: HintrAPIClient = mock {
+            on { downloadSpectrum("model1") } doReturn ResponseEntity.ok().body(StreamingResponseBody { it.write("".toByteArray()) })
+        }
+        val mockClient: ADRClient = mock {
+            on { postFile(eq("resource_patch"), eq(listOf("name" to "output1.zip", "description" to "Naomi model outputs", "hash" to "D41D8CD98F00B204E9800998ECF8427E", "resource_type" to "adr-output-zip", "id" to "resource1")), any()) } doReturn ResponseEntity.ok().body("whatever")
+            on { get("resource_show?id=resource1") } doReturn ResponseEntity.ok().body(objectMapper.writeValueAsString(data))
+        }
+        val mockBuilder: ADRClientBuilder = mock {
+            on { build() } doReturn mockClient
+        }
+        val sut = ADRController(mock(), mock(), mockBuilder, mock(), mockProperties, mock(), mockAPIClient, mock(), mock())
+        val hasOldHash = sut.uploadFileHasNoChanges("resource1", "D41D8CD98F00B204E9800998ECF8427E")
+        assertThat(hasOldHash).isTrue
+
+        val result = sut.pushFileToADR("dataset1", "adr-output-zip", "model1", "output1.zip", "resource1")
+        assertThat(result.statusCode).isEqualTo(HttpStatus.OK)
+        assertThat(objectMapper.readTree(result.body)["data"].textValue()).isEqualTo(null)
+    }
+
+    @Test
+    fun `update ADR if upload file has changes`()
```

Comment:
This seems to be testing the same thing as `pushes updated file to ADR`? If so then that test can be removed.